modules/service/src/main/scala/lucuma/odb/service/ProposalService.scala

```diff
@@ -567,15 +589,29 @@ object ProposalService {
             '{}'
           ) AS c_requested_partners,
           prog.c_goa_proprietary,
+          deadline.c_deadline,
+          (SELECT (deadline.c_deadline < CURRENT_TIMESTAMP):: boolean) AS c_is_past_deadline,
           cfp.c_cfp_id,
           cfp.c_type,
           cfp.c_semester,
           cfp.c_proprietary
         FROM t_program prog
         LEFT JOIN t_proposal prop
           ON prog.c_program_id = prop.c_program_id
-        LEFT JOIN t_cfp cfp
+        LEFT JOIN v_cfp cfp
           ON prop.c_cfp_id = cfp.c_cfp_id
+        LEFT JOIN t_program_user pi
+          ON prog.c_program_id = pi.c_program_id
+          AND pi.c_role = 'pi'
+        LEFT JOIN v_cfp_partner cfp_pi
+          ON cfp.c_cfp_id = cfp_pi.c_cfp_id
+          AND cfp_pi.c_partner = pi.c_partner
+        LEFT JOIN LATERAL
```

Comment:
By using `LATERAL` to calculate the deadline, I am also able to use it above to check if it is past the deadline.